client/state/data-layer/wpcom/sites/hosting/sftp-user.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import {
+	HOSTING_SFTP_USER_REQUEST,
+	HOSTING_SFTP_USER_CREATE,
+	HOSTING_SFTP_PASSWORD_RESET,
+} from 'state/action-types';
+import { errorNotice } from 'state/notices/actions';
+import { updateAtomicSftpUser } from 'state/hosting/actions';
+
+const requestAtomicSftpUser = action => {
+	return http(
+		{
+			method: 'GET',
+			path: `/sites/${ action.siteId }/hosting/ssh-user`,
+			apiNamespace: 'wpcom/v2',
+		},
+		action
+	);
+};
+
+const createAtomicSftpUser = action => {
+	return http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/ssh-user`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+};
+
+const resetAtomicSFTPPassword = action => {
+	return http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/ssh-user/reset-password`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+};
+
+const receiveAtomicSftpUserSuccess = ( action, { username, password } ) =>
+	updateAtomicSftpUser( action.siteId, action.userId, { username, password } );
+
+const sFTPUserError = ( { siteId, userId } ) => [
```

Comment:
Minor: 
```suggestion
const sftpUserError = ( { siteId, userId } ) => [
```

Reply:
haha, my global search and replace let me down yesterday :-)